projects/dsp-ui/src/lib/search/expert-search/expert-search.component.ts

```diff
@@ -0,0 +1,120 @@
+import { AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { KnoraApiConfig } from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken } from '../../core/core.module';
+import { AdvancedSearchParams, AdvancedSearchParamsService } from '../services/advanced-search-params.service';
+
+/**
+ * @ignore
+ * Validator checking that the query does not contain a certain term, here OFFSET
+ *
+ * @param {RegExp} termRe
+ */
+export function forbiddenTermValidator(termRe: RegExp): ValidatorFn {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+        const forbidden = termRe.test(control.value);
+        return forbidden ? { forbiddenName: { value: control.value } } : null;
+    };
+}
+
+@Component({
+    selector: 'dsp-expert-search',
+    templateUrl: './expert-search.component.html',
+    styleUrls: ['./expert-search.component.scss']
+})
+export class ExpertSearchComponent implements OnInit {
+
+    /**
+     * @param gravsearchQuery Send the gravsearch query back.
+     */
+    @Output() gravsearchQuery = new EventEmitter<string>();
+
+    expertSearchForm: FormGroup;
+    queryFormControl: FormControl;
+
+    defaultGravsearchQuery =
+        `PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+PREFIX incunabula: <${this.dspApiConfig.apiUrl}/ontology/0803/incunabula/simple/v2#>
+
+CONSTRUCT {
+    ?book knora-api:isMainResource true .
+    ?book incunabula:title ?title .
+
+} WHERE {
+    ?book a incunabula:book .
+    ?book incunabula:title ?title .
+}
+`;
+
+    constructor(
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig,
+        private _searchParamsService: AdvancedSearchParamsService,
+        private fb: FormBuilder
+    ) { }
+
+    ngOnInit(): void {
+        // initialize the form with predefined Gravsearch query as example.
+        this.queryFormControl = new FormControl(this.defaultGravsearchQuery);
+
+        this.expertSearchForm = this.fb.group({
+            gravsearchquery: [
+                this.defaultGravsearchQuery,
+                [
+                    Validators.required,
+                    forbiddenTermValidator(/OFFSET/i)
+                ]
+            ]
+        });
+    }
+
+    /**
+     * @ignore
+     * Reset the form to the initial state.
+     */
+    resetForm() {
+        this.expertSearchForm.reset({ gravsearchquery: this.defaultGravsearchQuery });
+    }
+
+    /**
+     * @ignore
+     * Send the gravsearch query to the result view by emitting the gravsearch as an output.
+     */
+    submitQuery() {
+        const gravsearch = this._generateGravsearch(0);
+
+        if (gravsearch) {
+            this.gravsearchQuery.emit(gravsearch);
+        }
+    }
+
+    /**
+     * @ignore
+     * Generate the whole gravsearch query matching the query given by the form.
+     */
+    private _generateGravsearch(offset: number = 0): string {
```

Comment:
I think you can assume that the offset is always 0 since the user is not allowed to provide an offset.

Reply:
so I can remove `= 0`?

Reply:
Please just create a DSP task so we can check this in

- expert-search
- gravsearch-generation-service

I think I should so that after my holiday ;-)

Reply:
discussion moved to [DSP-499](https://dasch.myjetbrains.com/youtrack/issue/DSP-499)